students/uss_tomasz/lesson_08_regular_expressions/8_2/strong_password_detection.py

```diff
@@ -0,0 +1,23 @@
+import re
+
+
+def is_password_strong(password):
+    length_regex = re.compile(r'[a-zA-Z0-9]{8,}')
+    if len(re.findall(length_regex, password)) == 0:
+        return False
+    lowercase_regex = re.compile(r'[a-z]')
+    if len(re.findall(lowercase_regex, password)) == 0:
+        return False
+    uppercase_regex = re.compile(r'[A-Z]')
+    if len(re.findall(uppercase_regex, password)) == 0:
+        return False
+    digits_regex = re.compile(r'\d')
+    if len(re.findall(digits_regex, password)) == 0:
+        return False
+    return True
```

Comment:
You can achieve the same using single regex formula with positive lookahead. You should see performance benefit for larger datasets.
https://stackoverflow.com/a/5142164